staging/src/k8s.io/apiserver/pkg/registry/rest/create.go

```diff
@@ -81,11 +81,17 @@ func BeforeCreate(strategy RESTCreateStrategy, ctx context.Context, obj runtime.
 		if !ValidNamespace(ctx, objectMeta) {
 			return errors.NewBadRequest("the namespace of the provided object does not match the namespace sent on the request")
 		}
-	} else {
+	} else if len(objectMeta.GetNamespace()) > 0 {
 		objectMeta.SetNamespace(metav1.NamespaceNone)
 	}
-	objectMeta.SetDeletionTimestamp(nil)
```

Comment:
can drop the deletion changes... the unstructured accessor already handles those correctly